fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -420,6 +420,11 @@ public Response createOrReplaceObjectRdf(
                 throw new GhostNodeException("Resource path " + externalPath() + " is an immutable resource.");
             }
 
+            if (!resourceExists && fedoraId.isDescription()) {
+                // Can't PUT a description to a non-existant binary.
+                throw new PathNotFoundException("Binary at path " + fedoraId.asBaseId().getFullIdPath() + " not found");
```

Comment:
This prints the following when you try to PUT to `/rest/fcr:metdata`:

> Error: Binary at path  not found

Maybe we should put a leading `/` or single quotes around it?

Reply:
Ok, I can add some simple handling for a repository root base ID.